src/kv/RocksDBStore.cc

```diff
@@ -1172,7 +1172,13 @@ void RocksDBStore::compact_thread_entry()
       logger->set(l_rocksdb_compact_queue_len, compact_queue.size());
       compact_queue_lock.Unlock();
       logger->inc(l_rocksdb_compact_range);
-      compact_range(range.first, range.second);
+      if (range.first.empty() && range.second.empty()) {
+        derr << "Begin to compact leveldb store all..." << dendl;
+        compact();
+        derr << "Finished to compact leveldb store all..." << dendl;
+      } else {
+        compact_range(range.first, range.second);
+      }
```

Comment:
this change is not relevant. please drop it.

Reply:
Do you mean this bug only impact leveldb but not rocksdb ?
Or I can use the compact_range function to do the full compact for both rocksdb and leveldb?

Reply:
no, i thought what we want to address is to make the command "compact" run in async, not to have make `compact_range_async("", "")` a special case. but yeah. you are right. we should